src/Core/Managed/Shared/OperationTelemetryExtensions.cs

```diff
@@ -87,8 +92,8 @@ public static void GenerateOperationId(this OperationTelemetry telemetry)
         private static void StopImpl(OperationTelemetry telemetry, long timestamp)
         {
             long stopWatchTicksDiff = timestamp - telemetry.BeginTimeInTicks;
-            double durationInMillisecs = stopWatchTicksDiff * 1000 / (double)Stopwatch.Frequency;
-            StopImpl(telemetry, TimeSpan.FromMilliseconds(durationInMillisecs));
+            double durationInTicks = stopWatchTicksDiff * StopwatchTicksToTimeSpanTicks;
```

Comment:
Great question. TL;DR: Yes. It's not an accident.

Notice that ```StopwatchTicksToTimeSpan``` is defined as a ratio. As it happens, on my machine (and, I think, most modern hardware), this ratio is exactly 1. However, on some hardware this ratio may be different and it's almost certainly not integral.

Now, you may note that, if you expand out the expression you get:
```durationInTicks = stopWatchTicksDiff * TimeSpan.TicksPerSecond / Stopwatch.Frequency```
in which all three terms on the right are integral (```long```) types. The danger here is with overflow. Since ```TimeSpan.TicksPerSecond``` is 10^7, that means the calculation will overflow when ```stopWatchTicksDiff``` is greater than 922,337,203,685. At the usual Stopwatch frequency, that corresponds to an elapsed time of ~25.6 hours which, while unusually long for an ASP.NET request, doesn't seem infeasible for long-lived batch processing operations.

So, to avoid the overflow, you want to divide down first and then we're back to the first point about the ratio needing to be non-integral.